nav2-usability/scenario_code/lib/deeplink-pathparam/deeplink_pathparam_beamer.dart

```diff
@@ -35,14 +36,18 @@ class BooksApp extends StatelessWidget {
             '/books/:bookId': (context) {
               final bookId = int.parse(
                   context.currentBeamLocation.state.pathParameters['bookId']!);
-              return BookDetailsScreen(
-                book: books[bookId],
+              return BeamPage(
+                key: ValueKey('book-$bookId'),
+                popToNamed: '/',
+                child: BookDetailsScreen(
```

Comment:
Why does `BookDetailsScreen` need to be wrapped within a `BeamPage` in the updated snippet?

Reply:
This is related to our [discussion at Discord](https://discord.com/channels/608014603317936148/813579416264507422/844001753518440458).

When pages are popped by default (`BackButton`; `Navigator.of(context).pop()`), Beamer needs to decide what the new URL will be. The default decision it will make is to remove the last path segment from URL. That would leave us with `/books`, but as this is not a supported path in the requirements (original `router` sample), we have 2 options;

- One is to use the `BeamGuard` as I described on Discord also, but this is somewhat too bulky and kind of inappropriate
```dart
BeamerRouterDelegate(
  guards: [
    BeamGuard(
      pathBlueprints: ['/', '/books/*'],
      guardNonMatching: true,
      check: (context, location) => false,
      beamToNamed: '/',
    ),
  ],
  ...
)
```
- The **real** solution for customized pops, which `v0.13` brings is to wrap the `Widget` with `BeamPage` which can take this custom pop either as `popToNamed` URI string or an `onPopPage` closure for even finer control (would be an overkill here).

Reply:
Got it, thanks.